--- gnuplot-orig/src/term.c	2006-12-27 16:40:28.000000000 -0500
+++ gnuplot/src/term.c	2007-06-15 00:35:51.000000000 -0400
@@ -2732,7 +2732,7 @@ on_page(int x, int y)
     if (term->flags & TERM_CAN_CLIP)
 	return TRUE;
 
-    if ((0 < x && x < term->xmax) && (0 < y && y < term->ymax))
+    if ((0 < x && x < term->xmax*xsize) && (0 < y && y < term->ymax*ysize))
 	return TRUE;
 
     return FALSE;
